File: src/main/java/eotree/data/EOFloatData.java
package eotree.data;

/**
 * EBNF representation:
 * <code>
 * /[+-]?\d+(\.\d+)?/
 * </code>
 */
public class EOFloatData extends EOData {
    public Float f;

    public EOFloatData(Float f) {
        this.f = f;
    }

    @Override
    public String generateEO(int indent) {
        return indent(indent) + f.toString();
    }
}

File: src/main/java/eotree/data/EOFloatData.kt
package eotree.data

import eotree.EONode

/**
 * EBNF representation:
 * `
 * /[+-]?\d+(\.\d+)?/
` *
 */
class EOFloatData(var f: Float) : EOData() {
    override fun generateEO(indent: Int): String {
        return indent(indent) + f.toString()
    }
}